### DISCUSSION TITLE
RuntimeError: expected scalar type Long but found Float

### DISCUSSION THREAD

**Discussion by garvpatidar04** [mrdbourke/pytorch-deep-learning#888] [Q&A] [answered]
epochs = 200

for epoch in range(epochs):
  model_3.train()

  y_logit = model_3(X_train1)

  loss = loss_fn1(y_logit, y_train1)

  optimizer.zero_grad()

  loss.backward()

  optimizer.step()

  model_3.eval()

  with torch.inference_mode():
    y_logit_t = model_3(X_test1)

    loss_test = loss_fn1(y_logit_t, y_test1)

    if epoch%40==0:
      print(f'Epoch: {epoch} | training loss: {loss} | test loss: {loss_test}') 
      
    above is the code for the training loop and I am getting a type error
    
![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/90318877/2b5ecefb-6de5-4182-a698-d3ba89fd5553)

and y_train1 is like this 
![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/90318877/1d3a7b8e-c807-413e-9c90-da924c3da457)

and y_logits is like this
![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/90318877/0a1a90ae-fc7e-4cb8-8f2f-8739cc010a16)

These are the optimizer and loss function
![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/90318877/8283cf85-3cad-4a47-a804-cfe96420e65e)

I have tried to change the dtype of y_train1 and y_logits, but it didn't work 

Can anyone help me to resolve the issue here, It would be a great help to me

**Comment by LuluW8071** [comment, marked as answer]
Typical datatype error
Just change `ytrain1` to long dtype. This is done for labels since labels should be fixed value not floating points.

Paste below snippet above your training loop code


```python
y_train1 = y_train1.type(torch.LongTensor)
```

**Reply by garvpatidar04** [reply]
Thank you, it did solve the problem, earlier I tried to change the type to `torch.long` and that didn't work because I changed it to inside the loop 😅.
